[stable/redis-ha] Allow for custom labels on redis pod

Francesco-Lanciana opened the pull request:
Signed-off-by: Francesco Lanciana <francescolanciana@gmail.com>

#### What this PR does / why we need it:
* Added the ability to assign extra labels to the redis pod (in addition to the release and app labels)

Currently there is no way to assign custom labels to the redis pod, however it is occasionally necessary for custom labels to be assigned to the pod. An example may be a label that is used to enforce a certain network security policy (this is our particular use case). Without this we are completely unable to proceed with this particular helm chart.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [ ] Variables are documented in the README.md

Francesco-Lanciana committed:
[stable/redis-ha] Allow for custom labels on redis pod

Signed-off-by: Francesco Lanciana <francescolanciana@gmail.com>

*What this PR does / why we need it:*
* Added the ability to assign extra labels to the redis pod (in addition to the release and app labels)

Currently there is no way to assign custom labels to the redis pod, however it is occasionally necessary for custom labels to be assigned to the pod. An example may be a label that is used to enforce a certain network security policy (this is our particular use case). Without this we are completely unable to proceed with this particular helm chart.

Signed-off-by: Francesco Lanciana <francescolanciana@gmail.com>

ssalaues commented on stable/redis-ha/templates/redis-ha-statefulset.yaml:
```diff
@@ -30,6 +30,9 @@ spec:
       labels:
         release: {{ .Release.Name }}
         app: {{ template "redis-ha.name" . }}
+        {{- range $key, $value := .Values.labels }}
```
Could you add a `labels: {}` line in the values.yaml as well?

ssalaues commented on stable/redis-ha/templates/redis-ha-statefulset.yaml:
```diff
@@ -30,6 +30,9 @@ spec:
         release: {{ .Release.Name }}
         app: {{ template "redis-ha.name" . }}
+        {{- range $key, $value := .Values.labels }}
+          {{ $key }}: {{ $value }}
```
Doesn't need to be indented more than 8 spaces

Francesco-Lanciana committed:
[stable/redis-ha] Updated the semver minor version

Signed-off-by: Francesco Lanciana <francescolanciana@gmail.com>

Updated the minor version for this chart as new functionality (the ability to add custom labels) was added to the chart. This is fully backwards compatible.

Signed-off-by: Francesco Lanciana <francescolanciana@gmail.com>

Francesco-Lanciana committed:
[stable/redis-ha] Added labels map to values.yaml

Signed-off-by: Francesco Lanciana <francescolanciana@gmail.com>

Added labels map to the values.yaml (base config for the chart)

Signed-off-by: Francesco Lanciana <francescolanciana@gmail.com>

Francesco-Lanciana committed:
[stable/redis-ha] Indentation for labels line in yaml no longer greater than 8 spaces

Signed-off-by: Francesco Lanciana <francescolanciana@gmail.com>

This line doesn't need to be indented more than 8 spaces

Signed-off-by: Francesco Lanciana <francescolanciana@gmail.com>
